Mathlib/Algebra/Order/Ring/Units.lean

```diff
@@ -0,0 +1,17 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Algebra.Ring.Subring.Units
+import Mathlib.GroupTheory.Index
```

Comment:
What exactly is the reason that this result isn't placed directly in `Mathlib.Algebra.Ring.Subring.Units`? Does importing `Mathlib.GroupTheory.Index` into that file violate some import condition or something? If so, I suggest that the correct thing to do is to actually move `Algebra.Ring.Subring.Units` to `Algebra.Order.Ring.Units`. (In fact, I think that's likely a reasonable idea anyway. That file doesn't really have anything to do with subrings.)

Reply:
The import `Cardinal` -> `Mathlib.GroupTheory.Index` -> `Mathlib.Algebra.Ring.Subring.Units` -> ... -> convex library is not liked by the `assert_not_exists Cardinal` in e.g. `Mathlib.Geometry.Convex.Cone.Basic`. I agree that `Mathlib.Algebra.Ring.Subring.Units` should be moved somewhere else but even if so it cannot be merged with this file.